include/vast/Dialect/HighLevel/HighLevelTypes.td

```diff
@@ -442,4 +457,10 @@ def AdjustedType : HighLevelType< "Adjusted" > {
   let assemblyFormat = "`<` $original `,` $adjusted `>`";
 }
 
+def ReferenceType : HighLevelType< "Reference" > {
```

Comment:
Can use the same helper as lvalue/rvalue. Also should implement `SubElementTypeInterface`.